### DISCUSSION TITLE
Default confidence and IoU threshold for training of yolov5

### DISCUSSION THREAD

**Discussion by rabiyaabbasi** [ultralytics/yolov5#7341] [Q&A]
Hi I wanted to know what are the default values of confidence and IoU threshold while training the data . I have checked train.py file but I am unable to find it. Secondly, attached below is my F1 score. according to which recall and precision optimises at a confidence value of 0.352. Do I need to use this confidence value for detection (!python detect.py --weights runs/train/exp4/weights/best.pt --img 416 --conf 0.352 --source ../dataset)? Secondly, when I checked the detect.py file, the IoU is 0.45 and confidence threshold is 0.25. Why we using this confidence threshold and how to update it?

![F1](https://user-images.githubusercontent.com/54917838/162274011-e55c522a-736c-4083-9c68-a476d58c36b9.png)

**Comment by glenn-jocher** [comment]
@rabiyaabbasi default settings are in detect.py argparser, you may modify these to whatever you want:
https://github.com/ultralytics/yolov5/blob/5f941a84efdd45c986cd1c3764ced99e7c8e8294/detect.py#L211-L243

**Reply by yesid-acm** [reply]
@glenn-jocher can you respond to the questions of @rabiyaabbasi ? Please! this is so important to understand the functionality.

**Reply by mahtabfaraji1** [reply]
@glenn-jocher can you respond to the questions of @rabiyaabbasi ? Thank you.
